### PR TITLE
wrf: fix patches for aarch64 config

### PR DESCRIPTION
**Pull request by annop-w**

### CODE REVIEW

**Review by MichaelLaufer — DISMISSED** [review]
@annop-w 
Thanks for this. 
Can you please make sure to submit the config for the Arm compiler to upstream WRF? This way we will not need to support these patches in future releases.
See https://github.com/wrf-model/WRF/pull/1684 for example.

File: var/spack/repos/builtin/packages/wrf/patches/4.2/add_aarch64.patch [hunk 1]
```diff
@@ -35,32 +35,29 @@ index 7767a81e..bfd2a802 100644
...
 +CFLAGS_LOCAL    =       -w -O3 -c -march=native -mtune=native
-+DFLAGS_LOCAL   =       
-+CPLUSPLUSLIB    =       
++DFLAGS_LOCAL   =
++CPLUSPLUSLIB    =
 +ESMF_LDFLAG     =       $(CPLUSPLUSLIB)
-+FCOPTIM         =       -O3 -ftree-vectorize -funroll-loops -march=native -mtune=native
-+FCREDUCEDOPT	=       $(FCOPTIM)
-+FCNOOPT		=       -O0
++FCOPTIM         =       -Ofast -fno-expensive-optimizations -fno-reciprocal-math -fsigned-zeros -fno-unsafe-math-optimizations -funroll-loops -march=native -mtune=native
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/wrf/patches/4.2/add_aarch64.patch]: These shouldn't be part of the optimizations used by the build system in Spack:
```suggestion
+FCOPTIM         =       -Ofast -fno-expensive-optimizations -fno-reciprocal-math -fsigned-zeros -fno-unsafe-math-optimizations -funroll-loops
```
If we use `-march=native` we might render the `target=...` on nodes meaningless.

Comment (annop-w) [reply var/spack/repos/builtin/packages/wrf/patches/4.2/add_aarch64.patch]: Done. Removed `-march` and `-mcpu`.

**Review by alalazo — APPROVED** [review]

**Comment by dodecatheon** [comment]
This patch breaks wrf@4.2 build with intel-oneapi-mpi@2021.8.0:
```
spack install wrf@4.2 build_type=dm+sm compile_type=em_real nesting=basic ~chem %gcc@9.3.0 ^intel-oneapi-mpi @2021.8.0
```

**Comment by alalazo** [comment]
@dodecatheon Can you post how it is breaking the build? Is it when applying the patch, or later during configuration or build?

**Comment by dodecatheon** [comment]
@alalazo :
```
==> Installing wrf-4.2-mpc5hi3fqydcqjg6ley27ghw2bffv22c
==> No binary for wrf-4.2-mpc5hi3fqydcqjg6ley27ghw2bffv22c found: installing from source
==> Using cached archive: /home/sternt_google_com/dev/spack/var/spack/cache/_source-cache/archive/c3/c39a1464fd5c439134bbd39be632f7ce1afd9a82ad726737e37228c6a3d74706.tar.gz
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/arch.Config.pl.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/arch.conf_tokens.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/arch.postamble.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/configure.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/external.io_netcdf.makefile.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/var.gen_be.Makefile.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/Makefile.patch
==> Applied patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/tirpc_detect.patch
Reversed (or previously applied) patch detected!  Assume -R? [n] n
Apply anyway? [n] y
1 out of 2 hunks FAILED -- saving rejects to file arch/configure.defaults.rej
==> Patch /home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/add_aarch64.patch failed.
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/patch' '-s' '-p' '1' '-i' '/home/sternt_google_com/dev/spack/var/spack/repos/builtin/packages/wrf/patches/4.2/add_aarch64.patch' '-d' '.'
    ```

**Pull request merged**